docs/gen_pages.py

```diff
@@ -0,0 +1,38 @@
+"""Generate the code reference pages."""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("gatorgrade").rglob("*.py")):  #
+    module_path = path.relative_to("gatorgrade").with_suffix("")  #
+    doc_path = path.relative_to("gatorgrade").with_suffix(".md")  #
+    full_doc_path = Path("reference", doc_path)  #
+
+    parts = list(module_path.parts)
+
+    if parts[-1] == "__init__":  #
+        parts = parts[:-1]
+    elif parts[-1] == "__main__":
+        continue
+
+    nav[parts] = doc_path.as_posix()
+
+    with mkdocs_gen_files.open(full_doc_path, "w") as fd:  #
+        identifier = "gatorgrade." + ".".join(parts)  #
+        print("::: " + identifier, file=fd)  #
```

Comment:
Using `fd.write` would make this code easier to understand.

Reply:
It's not actually a file object so you have to write to the file in this way. Refer to the documentation here: https://oprypin.github.io/mkdocs-gen-files/#usage